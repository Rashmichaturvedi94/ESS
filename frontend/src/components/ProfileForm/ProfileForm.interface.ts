import { FormProps } from '../../hooks/useForm.interface';

export interface ProfileFormValues {
  name: string;
  username: string;
  email: string;
  userId: number;
}

export type ProfileFormProps = FormProps<ProfileFormValues>;
