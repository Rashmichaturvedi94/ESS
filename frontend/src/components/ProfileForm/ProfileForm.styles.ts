import styled from 'styled-components';
import { TextField } from '@material-ui/core';

export const Profile = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
`;

export const NameField = styled(TextField)`
  height: 48px;
  margin: 4px 1px;
`;

export const UsernameField = styled(TextField)`
  height: 48px;
  margin-bottom: 4px;
`;

export const EmailField = styled(TextField)`
  height: 48px;
  margin: 4px 1px;
`;

export const PasswordField = styled(TextField)`
  height: 48px;
  margin-bottom: 4px;
`;
