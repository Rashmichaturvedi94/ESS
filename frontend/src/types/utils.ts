export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
