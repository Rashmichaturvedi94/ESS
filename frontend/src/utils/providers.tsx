import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
