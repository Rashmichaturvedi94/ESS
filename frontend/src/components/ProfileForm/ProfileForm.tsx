import React, { FC } from 'react';
import { Box, FormHelperText } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import * as Yup from 'yup';
import { NameField, UsernameField, EmailField } from './ProfileForm.styles';
import { useForm } from '../../hooks/useForm';
import { ProfileFormProps, ProfileFormValues } from './ProfileForm.interface';

const initialValues: ProfileFormValues = {
  name: '1',
  username: '',
  email: '',
  userId:0
};

const validationSchema: Yup.SchemaOf<ProfileFormValues> = Yup.object().shape({
  name: Yup.string().required('Required field'),
  username: Yup.string().required('Required field'),
  email: Yup.string().required('Required field'),
  userId: Yup.number().required('Required field'),
});

export const ProfileForm: FC<ProfileFormProps> = (props) => {
  const { errors, values, handleChange, handleSubmit } = useForm({
    initialValues,
    validationSchema,
    ...props,
  });

  return (
    <form onSubmit={handleSubmit}>
      <Box display="flex" flexDirection="column" gridRowGap={20}>
        {errors?.detail && (
          <FormHelperText error>{errors?.detail}</FormHelperText>
        )}
        <NameField
          fullWidth
          name="name"
          placeholder='Name'
          error={!!errors.name}
          helperText={errors.name}
          value={values.name}
          disabled
        />
        <UsernameField
          fullWidth
          name="username"
          placeholder='Username'
          error={!!errors.username}
          helperText={errors.username}
          value={values.username}
          disabled
        />
        <EmailField
          fullWidth
          name="email"
          placeholder='Email'
          error={!!errors.email}
          helperText={errors.email}
          value={values.email}
          onChange={handleChange}
        />
        <Button type="submit" variant="contained" fullWidth>
          Save
        </Button>
      </Box>
    </form>
  );
};
