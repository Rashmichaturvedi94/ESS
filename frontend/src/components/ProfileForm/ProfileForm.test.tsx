import React from 'react';
import { render } from '../../utils/testUtils';
import { ProfileForm } from './ProfileForm';

describe('components/ProfileForm', () => {
  it('should render', () => {
    const { container } = render(<ProfileForm />);
    expect(container).toMatchSnapshot();
  });
});
