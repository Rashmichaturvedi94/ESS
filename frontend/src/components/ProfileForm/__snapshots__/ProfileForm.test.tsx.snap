// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProfileForm should render 1`] = `
<div>
  <form>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-dlnjwi cqBiuT MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
            disabled=""
            name="username"
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root sc-gtsrHT bdAwyA MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="name"
            placeholder="Name"
            type="text"
            value="1"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root sc-hKFxyN GcSgM MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="email"
            placeholder="Email"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Save
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
